pkg/datagatherer/istio/istio_test.go

```diff
@@ -0,0 +1,166 @@
+package istio
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"testing"
+
+	"gopkg.in/yaml.v2"
+)
+
+const tempFilePrefix = "preflight-test-istio-datagatherer"
+
+// TestFetch runs a full test of the Istio data gatherer; running a fake Kubernetes API, using dynamic data gatherers to
+// fetch resources, running fetched resources through Istio analysis, and checking that the analysis messaged generated
+// are what was expected.
+func TestFetch(t *testing.T) {
+	// Local server to handle requests made by Kubernetes dynamic data gatherer.
```

Comment:
As I see this, the reason we need a localServer for this rather than using the fake client is that we have no way to inject the fake client to these 'level 2' datagathers which use the dynamic datagatherer as a dependency (version checker being the other datagatherer of the same level).

I'm happy with using a local server and feel that it's what I might have reached for when testing the dynamic dg originally if I worked on it.

However, I think that this is worth explaining in a comment as to why it's different from the tests for the dynamic dg.

Reply:
I made a comment in the version checker dg related to this here: #218

Reply:
I've added an explanation for this 👍